app/controllers/idv/phone_errors_controller.rb

```diff
@@ -45,6 +46,11 @@ def confirm_idv_phone_step_needed
       redirect_to idv_review_url if idv_session.user_phone_confirmation == true
     end
 
+    def confirm_idv_phone_step_submitted
+      rate_limiter = RateLimiter.new(user: current_user, rate_limit_type: :proof_address)
```

Comment:
Interesting idea to use RateLimiter! We've been checking [idv_session](https://github.com/18F/identity-idp/blob/1e09def9ae8fec2243cf395e3d37164aaf5c97ae/app/services/idv/session.rb#L157-L159) for statuses like this. Do we want to adopt RateLimiter as a new pattern?

Reply:
you're right  😅 ... i have reimplemented this using the idv_session ...thanks @soniaconnolly   👍🏿